### PR TITLE
Allow `unused_parens` temporarily

### PR DESCRIPTION
**Pull request by JohnTitor**
It's annoying on CI so we should allow this lint temporarily, I think.

### CODE REVIEW

**Comment by weiznich** [comment]
I would rather work on fixing that in rustc. The critical line in rustc is [that](https://github.com/rust-lang/rust/blob/a44774c3a9739b2eea8923e09d67b14312c78ef3/src/librustc_lint/unused.rs#L358-L359) one. If someone knows how to check there if we are in a macro expansion it should be simple to be fixed.

**Comment by JohnTitor** [comment]
This is just a stopgap. Actually, the lint errors hide test failure so it's worth to avoid the lint failure, for now, I think. And it will take some time until the fix lands on beta unless backport.

File: diesel_tests/tests/joins.rs [hunk 1]
```diff
@@ -678,13 +678,13 @@ fn connection_with_fixture_data_for_multitable_joins() -> (TestConnection, TestD
         .order(posts::id)
         .load::<Post>(&connection)
         .unwrap();
-    let new_comments = vec![
+    let new_comments: &[NewComment<'static>] = &[
         NewComment(posts[0].id, "First Comment"),
```
Comment (weiznich) [inline diesel_tests/tests/joins.rs]: Any idea why this fails on nightly? Should we maybe report a regression against rustc?

Comment (JohnTitor) [reply diesel_tests/tests/joins.rs]: I have no idea. We haven't changed behavior for it and the failure only appears on nightly so I guess recent changes on nightly cause the failure (I'm not sure it's a regression or expected behavior though).

Comment (weiznich) [reply diesel_tests/tests/joins.rs]: I've retriggered the ci. Seems to be fixed now. You need to run rustfmt.

**Pull request merged**